mesonbuild/backend/backends.py

```diff
@@ -1263,12 +1263,18 @@ def create_test_serialisation(self, tests: T.List['Test']) -> T.List[TestSeriali
 
             t_env = copy.deepcopy(t.env)
             if not machine.is_windows() and not machine.is_cygwin() and not machine.is_darwin():
-                ld_lib_path: T.Set[str] = set()
+                ld_lib_path_libs: T.Set[build.SharedLibrary] = set()
                 for d in depends:
                     if isinstance(d, build.BuildTarget):
                         for l in d.get_all_link_deps():
                             if isinstance(l, build.SharedLibrary):
-                                ld_lib_path.add(os.path.join(self.environment.get_build_dir(), l.get_subdir()))
+                                ld_lib_path_libs.add(l)
+
+                env_build_dir = self.environment.get_build_dir()
+                ld_lib_path: T.Set[str] = set()
+                for l in ld_lib_path_libs:
+                    ld_lib_path.add(os.path.join(env_build_dir, l.get_subdir()))
```

Comment:
If we know that lib_path_libs is already unique then maybe we can just construct the list which we'd use a few lines down in t_env.prepend, directly?

Reply:
Using something like `map`? Sure, on it.

Reply:
Oh, never mind, I see what you mean.

Reply:
The easiest way to simplify this and get rid of the two sets would be to store the subdirs instead of the targets in the initial set, which was the original version of this patch, but it still ran get_subdir much more than necessary and py-spy showed that python was spending a lot of time in that function. I can simplify with a list comprehension, but I think beyond that would be a step backwards.

Reply:
Yeah, my bad, I think I misread. Deduplicating twice makes sense indeed.

Reply:
Updated the commit with list comprehension for the latter set's construction, and additional information in the commit messages.